src/main/java/jp/ikedam/jenkins/plugins/updatesitesmanager/UpdateSitesManager.java

```diff
@@ -23,274 +23,205 @@
-     * </ol>
-     * 
-     * @param req
-     * @param rsp
-     * @return
-     * @throws IOException
-     * @throws UnsupportedEncodingException
-     * @throws ServletException
-     * @throws FormException
+     * Update all registered sites with concatenation of managed and not managed
+     * @param managed managed sites form submitted form
+     * @return redirect to same url if all ok
      */
-    public HttpResponse do_add(StaplerRequest req, StaplerResponse rsp) throws IOException, UnsupportedEncodingException, ServletException, FormException
-    {
-        // Only administrator can create a new site.
-        Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
-        
-        Descriptor<DescribedUpdateSite> descriptor = null;
-        
-        List<DescribedUpdateSite.Descriptor> descriptorList = getUpdateSiteDescriptorList();
-        
-        if(req.getParameter("stapler-class") != null)
-        {
-            try
-            {
-                @SuppressWarnings("unchecked")
-                Class<DescribedUpdateSite.Descriptor> clazz = (Class<DescribedUpdateSite.Descriptor>)Class.forName(req.getParameter("stapler-class"));
-                descriptor = Jenkins.getInstance().getDescriptorByType(clazz);
-            }
-            catch(ClassCastException e)
-            {
-                LOGGER.log(Level.WARNING, String.format("Bad stapler-class: %s", req.getParameter("stapler-class")), e);
-            }
-            catch (ClassNotFoundException e)
-            {
-                LOGGER.log(Level.WARNING, String.format("Bad stapler-class: %s", req.getParameter("stapler-class")), e);
-            }
-            
-        }
-        
-        if(descriptor == null && descriptorList.size() == 1)
-        {
-            descriptor = descriptorList.get(0);
-        }
-        
-        if(descriptor == null)
-        {
-            // There are more than one DescribedUpdateSite is available,
-            // so let the user to select which to create.
-            return new ForwardToView(this, "newSiteSelect.jelly");
-        }
-        
-        if(!"POST".equals(req.getMethod()))
-        {
-            // Show form.
-            req.setAttribute("instance", null);
-            req.setAttribute("descriptor", descriptor);
-            return new ForwardToView(this, "newSite.jelly");
+    @OnlyAdminister
+    @IgnoreNotPOST
+    @SuppressWarnings("unused")
+    public HttpResponse doUpdate(@Sites List<UpdateSite> managed) throws ServletException, IOException, FormException {
+        List<UpdateSite> newSitesList = newArrayList(Iterables.concat(getNotManagedUpdateSiteList(), managed));
+
+        shouldNotContainDuplicatedIds(newSitesList);
+        shouldNotContainBlankIds(newSitesList);
+
+        Jenkins.getInstance().getUpdateCenter().getSites().replaceBy(newSitesList);
+        Jenkins.getInstance().getUpdateCenter().save();
+
+        return HttpResponses.redirectToDot();

```

Comment:
I think that it should be some kind of verification for user that all saved without any problem. So i don't know on how to make *apply* btn and make combined behaviour in the *save* btn